static/js/components/PublishDrawer.tsx

```diff
@@ -1,151 +1,160 @@
 import React, { useState } from "react"
 import { Modal, ModalBody, ModalHeader } from "reactstrap"
-import { useStore } from "react-redux"
 import { useMutation } from "redux-query-react"
 import { requestAsync } from "redux-query"
 import moment from "moment"
 import { isEmpty } from "ramda"
 
 import {
-  websiteAction,
+  websitePublishAction,
   websiteDetailRequest,
   WebsitePublishPayload
 } from "../query-configs/websites"
 import { isErrorStatusCode } from "../lib/util"
 import PublishStatusIndicator from "./PublishStatusIndicator"
 
 import { Website } from "../types/websites"
-import { PublishForm } from "./forms/PublishForm"
-import { PUBLISH_OPTION_PRODUCTION, PUBLISH_OPTION_STAGING } from "../constants"
+import PublishForm, {
+  OnSubmitPublish,
+  SiteFormValues
+} from "./forms/PublishForm"
+import { PublishingEnv, PublishStatus } from "../constants"
+import { useAppDispatch } from "../hooks/redux"
 
-type Props = {
-  visibility: boolean
-  toggleVisibility: () => void
+interface PublishingOptionProps {
   website: Website
+  publishingEnv: PublishingEnv
+  selected: boolean
+  onSelect: (publishingEnv: PublishingEnv) => void
+  onPublishSuccess: () => void
+}
+interface PublishingInfo {
+  date: string | null
+  status: PublishStatus | null
+  hasUnpublishedChanges: boolean
+  envLabel: string
 }
-export default function PublishDrawer(props: Props): JSX.Element {
-  const { visibility, toggleVisibility, website } = props
 
-  const [
-    { isPending: previewIsPending },
-    previewWebsite
-  ] = useMutation((payload: WebsitePublishPayload) =>
-    websiteAction(website.name, "preview", payload)
-  )
+const getPublishingInfo = (
+  website: Website,
+  publishingEnv: PublishingEnv
+): PublishingInfo => {
+  if (publishingEnv === PublishingEnv.Staging) {
+    return {
+      envLabel:              "Staging",
+      date:                  website.draft_publish_date,
+      status:                website.draft_publish_status,
+      hasUnpublishedChanges: website.has_unpublished_draft
+    }
+  }
+  if (publishingEnv === PublishingEnv.Production) {
+    return {
+      envLabel:              "Production",
+      date:                  website.publish_date,
+      status:                website.live_publish_status,
+      hasUnpublishedChanges: website.has_unpublished_live
+    }
+  }
+  throw new Error("Invalid PublishingEnv")
+}
+
+const PublishingOption: React.FC<PublishingOptionProps> = props => {
```

Comment:
This was previously a function defined in `PublishingDrawer`, `renderOption`, now a real component.